sigle/src/modules/analytics/stats/StatsFrame.tsx

```
@@ -0,0 +1,178 @@
+
+  useEffect(() => {
+    fetchWeeklyStats();
+  }, []);
+
+  const fetchWeeklyStats = async () => {
+    setData(initialRange);
+
+    const statsRes = await fetch(
+      `${baseUrl}/api/analytics/historical?dateFrom=${format(
+        weekFromDate,
+        'yyyy-MM-dd'
+      )}&dateGrouping=day`
+    );
+
+    const statsData: AnalyticsHistoricalResponse = await statsRes.json();
+    const stats: StatsData[] = statsData.historical.map((item) => {
+      return {
+        value: item.pageviews,
+        date: item.date,
+        visits: item.visits,
+      };
+    });
+
+    setData(stats);
+  };
+
+  const fetchMonthlyStats = async () => {
+    setData(initialRange);
+
+    const statsRes = await fetch(
+      `${baseUrl}/api/analytics/historical?dateFrom=${format(
+        monthFromDate,
+        'yyyy-MM-dd'
+      )}&dateGrouping=day`
+    );
+
+    const statsData: AnalyticsHistoricalResponse = await statsRes.json();
+    const stats: StatsData[] = statsData.historical.map((item) => {
+      return {
+        value: item.pageviews,
+        date: item.date,
+        visits: item.visits,
+      };
+    });
+    setData(stats);
+  };
+
+  const fetchAllTimeStats = async () => {
+    setData(initialRange);
+
+    const statsRes = await fetch(
+      `${baseUrl}/api/analytics/historical?dateFrom=${FATHOM_MAX_FROM_DATE}&dateGrouping=month`
+    );
+
+    const statsData: AnalyticsHistoricalResponse = await statsRes.json();
+    const stats: StatsData[] = statsData.historical.map((item) => {
+      return {
+        value: item.pageviews,
+        date: item.date,
+        visits: item.visits,
+      };
+    });
+
+    setData(stats);
+  };
+
+  const fetchStats = (value: string) => {
+    switch (value) {
+      case 'weekly':
+        fetchWeeklyStats();
+        break;
+      case 'monthly':
+        fetchMonthlyStats();
+        break;
+      case 'all':
+        fetchAllTimeStats();
+        break;
+      default:
+        fetchWeeklyStats();
```

Default should probably throw an error as this should not happen :)